### PR TITLE
Bug fix and add function to make mp4

### PR DESCRIPTION
**Pull request by romanlee**

### CODE REVIEW

**Comment by Wen-Han** [comment]
I have a question about the numpy.linspace change. Can you provide an example of the problem you encountered with numpy.arange?

Other parts of the code is tied to the numpy.range. For example, DataAxis.max will return (N+1)*dx instead of N*dx. If this commit go through, other parts have to be modified accordingly.

**Comment by romanlee** [comment]
Hey Han,

Good point.  The problem I was having was with using osplot1d() to plot
phasespace data.  I read in some hdf5 file, call it c, then call
osh5vis.osplot1d( c ).  Then I would get the following error

ValueError: x and y must have same first dimension, but have shapes (101,)
and (100,)


The problem was that the number of points on self.ax in line 25 os
ohs5def.py would be different, because of floating point stuff, than the
number of points of h5data on line 102 of osh5vis.  If you use
np.linspace() instead of np.arange() you ensure that np.shape(axis) =
np.shape(h5data), which fixes the bug.

I'm attaching a little zip of with a script and a couple files that give
the bug so you can see for yourself.  Just make sure you import your
library inside of debug.py.

Let me know what you think.

Roman

**Comment by Wen-Han** [comment]
Hi Roman,

I checked the osiris source code. I think the numpy.arange will give us the correct grid size (dx). We have to address the numerical issue you had also. I suggest we do the following:
`self.ax = np.arange(axis_min, axis_max - (axis_max - axis_min) / axis_npoints, axis_npoints)`

We only need to change this one place. If this looks good to you can  you make the changes? I will merge it when you are done.

Thanks,
Han

File: osh5vis.py [hunk 1]
```diff
@@ -250,3 +251,23 @@ def new_fig(h5data, *args, figsize=None, dpi=None, facecolor=None, edgecolor=Non
     osplot(h5data, *args, **kwpassthrough)
     plt.show()
 
+
+def movie( scan_dir, fname, fps=50, fig_size=[50,20], dpi=120 ):
```
Comment (Wen-Han) [inline osh5vis.py]: Always use immutable as default paramters (change list to tuple):
`def movie( scan_dir, fname, fps=50, fig_size=(50,20), dpi=120 )`

File: osh5vis.py [hunk 2]
```diff
@@ -250,3 +251,23 @@ def new_fig(h5data, *args, figsize=None, dpi=None, facecolor=None, edgecolor=Non
...
+    :dpi = image resolution
+    :fig_size = fig_size
+    """
+    f = scan_dir + fname
+
+    x = dpi * fig_size[0]
+    y = dpi * fig_size[1]
+    if(x*y > 4000 * 2000):
+        x = x/2
+        y = y/2
```
Comment (Wen-Han) [inline osh5vis.py]: I think libx264 requires the width and height to be divisible by 2, can you confirm that? For example fig_size=(51,21), dpi=121 will not work with libx264.

An easy solution would be before calling ffmpeg do the following:
`x, y = x // 2 * 2, y // 2 * 2`

Comment (romanlee) [reply osh5vis.py]: Yes, you're correct.  I'll make that change.

File: osh5def.py [hunk 3]
```diff
@@ -22,7 +22,7 @@ def __init__(self, axis_min=0., axis_max=1., axis_npoints=1, attrs=None, data=No
         if data is None:
             if axis_min > axis_max:
                 raise Exception('illegal axis range: [ %(l)s, %(r)s ]' % {'l': axis_min, 'r': axis_max})
-            self.ax = np.arange(axis_min, axis_max, (axis_max - axis_min) / axis_npoints)
+            self.ax = np.linspace(axis_min, axis_max, axis_npoints)
```
[the remark this answers is not in the document]
Comment (romanlee) [reply osh5def.py]: This looks good to me.

**Pull request merged**